src/main/java/com/codeborne/selenide/selector/ByShadow.java

```diff
@@ -66,13 +68,31 @@ public WebElement findElement(SearchContext context) {
     @Override
     @CheckReturnValue
     @Nonnull
-    public List<WebElement> findElements(SearchContext context) {
-      WebElement host = context.findElement(By.cssSelector(shadowHost));
-      for (String innerHost : innerShadowHosts) {
-        host = getElementInsideShadowTree(host, innerHost);
+    public List<WebElement> findElements(final SearchContext context) {
+      final List<WebElement> hosts = context.findElements(By.cssSelector(shadowHost));
+
+      final List<WebElement> innerHosts = findInnerShadowHosts(hosts, Arrays.asList(this.innerShadowHosts));
+
+      return innerHosts.stream()
+        .flatMap(host -> getElementsInsideShadowTree(host, this.target).stream())
+        .collect(Collectors.toList());
+    }
+
+    private List<WebElement> findInnerShadowHosts(final List<WebElement> shadowHosts,
+                                                  final List<String> innerHostSelectors) {
+      List<WebElement> hosts = shadowHosts;
+      List<String> hostSelectors = innerHostSelectors;
+      while (!hostSelectors.isEmpty()) {
+        final List<WebElement> innerHosts = new ArrayList<>();
+        final String innerHostSelector = innerHostSelectors.get(0);
```

Comment:
Should not we get the inner host from `hostSelectors`?

Reply:
good catch. you're right of course. I'm going to fix this.

Reply:
See updates in the pull request